cdap-kubernetes/src/main/java/io/cdap/cdap/k8s/runtime/KubeTwillController.java

```diff
@@ -526,36 +536,24 @@ private void deletePVCs() {
   private CompletionStage<String> deleteJobAndGetCompletionStatus() {
     CompletableFuture<String> resultFuture = new CompletableFuture<>();
     String name = meta.getName();
-    try {
-      V1Job v1Job = batchV1Api.readNamespacedJob(name, kubeNamespace, null, null, null);
-      V1JobStatus status = v1Job.getStatus();
-
+    if (jobStatus == null) {
+      isStopped = true;
+      resultFuture.complete(name);
```

Comment:
If we return here, we don't need `else` in the next line.

Reply:
We still need to attempt to delete the job.